File: problem_039.py
#!/usr/bin/env python
import math

print("Project Euler - Problem 39:")
print("Find the perimeter of a right triangle (p) that has the most possible solutions, where the perimiter is less than or equal to 1000.\n")

limit = 1000/2
perimeter = {}

for a in range (1, limit):
    for b in range (a, limit):
        c =  math.sqrt((a*a) + (b*b))
        if c < limit:
            if c == int(c):
                sum = a + b + int(c)
                if not sum in perimeter:
                    perimeter.update({sum:0})
                perimeter[sum] += 1

print(sorted(perimeter, key=perimeter.get)[-1])

File: problem_039.js
#!/usr/bin/node

console.log("Project Euler - Problem 39:");
console.log("Find the perimeter of a right triangle (p) that has the most possible solutions, where the perimiter is less than or equal to 1000.\n");

var solutions = {};
var limit = 1000;

for (var a=1; a<limit; a++){
    for (var b=1; a+b<=limit; b++){
        let c = Math.sqrt((a * a) + (b * b));
        if (Math.floor(c) == c){
            let p = a + b + c;
            if (p > limit) break;
            if (p in solutions) {
                solutions[p]++;
            } else {
                solutions[p] = 1;
            }
        }
    }
}

var max = null;
for (var i in solutions)
    if (solutions[i] > solutions[max] || max == null)
        max = i;

console.log(max);
